rxsample/src/androidTest/java/com/mateuszkoslacz/moviper/rxsample/utils/di/UserTestRepository.java

```diff
@@ -0,0 +1,78 @@
+package com.mateuszkoslacz.moviper.rxsample.utils.di;
+
+import android.util.Log;
+
+import com.mateuszkoslacz.moviper.rxsample.data.rdp.specification.list.base.Specification;
+import com.mateuszkoslacz.moviper.rxsample.data.rdp.specification.stream.base.StreamSpecification;
+import com.mateuszkoslacz.moviper.rxsample.viper.entity.User;
+
+import java.util.List;
+
+import rx.Observable;
+import rx.schedulers.TestScheduler;
+import rx.subjects.TestSubject;
+
+
+public class UserTestRepository implements TestRepository<User> {
+
+    private TestScheduler testScheduler;
+    private TestSubject<List<User>> testUsersSubject;
+
+    public UserTestRepository() {
+        Log.d("UserTestRepository", "constructor");
+        this.testScheduler = new TestScheduler();
+        this.testUsersSubject = TestSubject.create(testScheduler);
+    }
+
+    @Override
+    public void add(User item) {
+
+    }
+
+    @Override
+    public void update(User item) {
+
+    }
+
+    @Override
+    public void remove(User item) {
+
+    }
+
+    @Override
+    public Observable<User> streamQuery(StreamSpecification specification) {
+        return null;
+    }
+
+    @Override
+    public Observable<List<User>> query(Specification specification) {
+        Log.d("UserTestRepository", "query");
+        return testUsersSubject;
+    }
+
+    @Override
+    public void triggerContentReturn() {
```

Comment:
It makes no sense as the Specification is the object that originally returns the data. Moreover, you have implemented it there already. In fact, applying the suggestions below, here you could do `testScheduler.triggerActions()` only.